wit/types.wit

```diff
@@ -34,6 +34,16 @@ interface types {
     HTTPS,
     other(string)
   }
+  }
+
+  /// This type corresponds to HTTP protocol versions.
+  @since(version = 0.2.2)
+  variant http-version {
+    HTTP11,
+    HTTP2,
+    HTTP3
+    other(string)
```

Comment:
Adding HTTP4 here would be API breaking change. 

There is similar discussion about TSL version, with regards to future compatibility. 
https://github.com/WebAssembly/wasi-sockets/pull/104#discussion_r1725442743

Reply:
Yeah, I saw that. My intuition is that HTTP versions are introduced rarely enough that we'll have solved this by the time HTTP/4 comes around, but I'm happy to change it if people feel that that's not right.